File: app/src/main/java/gs/weather/engine/Utility.java
package gs.weather.engine;

import android.util.Log;

public class Utility {
    static boolean DEBUG = false;

    public static class Logger {
        public static void v(String tag, String info) {
            if (Utility.DEBUG) {
                Log.v(tag, info);
            }
        }

        public static void d(String tag, String info) {
            if (Utility.DEBUG) {
                Log.d(tag, info);
            }
        }

        public static void i(String tag, String info) {
            if (Utility.DEBUG) {
                Log.i(tag, info);
            }
        }

        public static void w(String tag, String info) {
            if (Utility.DEBUG) {
                Log.w(tag, info);
            }
        }

        public static void e(String tag, String info) {
            if (Utility.DEBUG) {
                Log.e(tag, info);
            }
        }
    }

    public static void adjustScreenPosForDepth(Vector destVector, float cameraFOV, float screenWidth, float screenHeight, float touchX, float touchY, float depth) {
        float f10 = (cameraFOV * (screenWidth / screenHeight)) * 0.01111111f;
        float f14 = (cameraFOV * 0.01111111f) * ((((1.0f - (touchY / screenHeight)) - 0.5f) * 2.0f) * depth);
        destVector.setX(((((touchX / screenWidth) - 0.5f) * 2.0f) * depth) * f10);
        destVector.setY(depth);
        destVector.setZ(f14);
    }

}

File: app/src/main/kotlin/io/github/gmazzo/android/livewallpaper/weather/engine/Utility.kt
package io.github.gmazzo.android.livewallpaper.weather.engine

import android.util.Log

object Utility {
    var DEBUG: Boolean = true

    fun adjustScreenPosForDepth(
        destVector: Vector,
        cameraFOV: Float,
        screenWidth: Float,
        screenHeight: Float,
        touchX: Float,
        touchY: Float,
        depth: Float
    ) {
        val f10 = (cameraFOV * (screenWidth / screenHeight)) * 0.01111111f
        val f14 =
            (cameraFOV * 0.01111111f) * ((((1.0f - (touchY / screenHeight)) - 0.5f) * 2.0f) * depth)
        destVector.x = ((((touchX / screenWidth) - 0.5f) * 2.0f) * depth) * f10
        destVector.y = depth
        destVector.z = f14
    }

    object Logger {
        fun v(tag: String?, info: String?) {
            if (DEBUG) {
                Log.v(tag, info!!)
            }
        }

        fun d(tag: String?, info: String?) {
            if (DEBUG) {
                Log.d(tag, info!!)
            }
        }

        fun i(tag: String?, info: String?) {
            if (DEBUG) {
                Log.i(tag, info!!)
            }
        }

        fun w(tag: String?, info: String?) {
            if (DEBUG) {
                Log.w(tag, info!!)
            }
        }

        fun e(tag: String?, info: String?) {
            if (DEBUG) {
                Log.e(tag, info!!)
            }
        }
    }
}
